src/main/java/jarvis/ui/LessonCard.java

```diff
@@ -50,12 +53,19 @@ public class LessonCard extends UiPart<Region> {
     public LessonCard(Lesson lesson, int displayedIndex) {
         super(FXML);
         this.lesson = lesson;
+        hasClash.textProperty().addListener((unused1, unused2, unused3) -> setStyleForClash());
```

Comment:
just curious, so the red colour for the card is set when the invisible text in `hasClash` is changed to "Clash", is it implemented this way because its a convenient way trigger the red colour ?

Reply:
I had a similar concern, it seems like `hasClash` is a UI specific property rather than an attribute of the actual Lesson. Is it possible to shift it to the UI component to increase cohesion of the Model classes?